core/src/org/pentaho/di/core/encryption/CertificateGenEncryptUtil.java

```diff
@@ -53,10 +62,52 @@ public static KeyPair generateKeyPair() {
     return pair;
   }
 
+  public static Key generateSingleKey() throws NoSuchAlgorithmException {
+    Key key = KeyGenerator.getInstance( SINGLE_KEY_ALGORYTHM ).generateKey();
+    return key;
+  }
+
+  public static byte[] encodeKeyForTransmission( Key encodingKey, Key keyToEncode ) throws Exception {

```

Comment:
Let's throw the more specific exceptions here, then test for some of the more likely ones in the unit tests.